src/utils/vidosResolver.ts

```diff
@@ -0,0 +1,43 @@
+import type {
+    DIDResolutionOptions,
+    DIDResolutionResult,
+    Resolvable
+} from "did-resolver";
+
+/** 
+ * Create resolver that is using the Vidos service to resolve DIDs. 
+ * 
+ * - For more info: [https://vidos.id/](https://vidos.id/).
+ * - For resolver configuration: [https://dashboard.vidos.id/](https://dashboard.vidos.id/).
+ * - For docs: [https://vidos.id/docs/](https://vidos.id/docs/).
+ */
+export class VidosResolver implements Resolvable {
+    constructor(readonly instanceUrl: string, readonly apiKey: string) {}
+  
+    async resolve(
+        didUrl: string,
+        options?: DIDResolutionOptions | undefined
+    ): Promise<DIDResolutionResult> {
+        const resolutionResponse = await fetch(`${this.instanceUrl}/${didUrl}`, {
+            method: "GET",
+            headers: {
+                Authorization: `Bearer ${this.apiKey}`,
+                Accept: options?.accept ?? "",
+            },
+        });
+  
+        if (resolutionResponse.ok) return resolutionResponse.json();
```

Comment:
Soon to be implemented but many of the HTTP error codes will also return a valid DIDResolutionResult. Look like with this check it will not be handled correctly